Encapsulated support (RTU over TCP) (updated #260)

v-zhuravlev opened the pull request:
Hello, updated this pull request https://github.com/stephane/libmodbus/pull/260 to 3.1.4 and main branch(as #260 was based on 3.0.3 as far as I see it). 

It would be great if encapsulated is supported in the main branch somehow.

Thanks.

v-zhuravlev committed:
added encapsulated to 3.1.4, works with libzbxmodbus.

v-zhuravlev committed:
small clean up in headers

v-zhuravlev committed:
whitespace and comments cleanup

fatkhrus commented:
It will be great if encapsulated will be supported in the main branch

evgkrsk commented:
ping?

strahlc commented:
why is this not merged?
I need this patch 😄

kiug commented:
Me too

25 sty 2018 4:36 PM "Christian Strahl" <notifications@github.com>
napisał(a):

> why is this not merged?
> I need this patch 😄
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/stephane/libmodbus/pull/385#issuecomment-360503192>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAK-69w8rjuyttd45AOgFjWwzPqoSV-Uks5tOJ-LgaJpZM4Ns09C>
> .
>

bashlakov commented:
This patch is very useful when using RS485 over IP converters without installing virtual com-port software, can it be merged in mainline?

pboettch commented:
@v-zhuravlev I think it would help the integration (if Stephane finds the time) if there would a test for this functionality. Having said that, I know that, I wouldn't know how to test as it seems to be only a client-side implementation.

And documentation for the rtutcp-function is needed as well.

Maybe with the `_raw`-function somehow.

v-zhuravlev commented:
I’m not sure that I’m capable to do all that properly. But I m ready to help if given proper guideance.

Отправлено с телефона

> 15 февр. 2018 г., в 17:38, Patrick Boettcher <notifications@github.com> написал(а):
> 
> @v-zhuravlev I think it would help the integration (if Stephane finds the time) if there would a test for this functionality. Having said that, I know that, I wouldn't know how to test as it seems to be only a client-side implementation.
> 
> And documentation for the rtutcp-function is needed as well.
> 
> Maybe with the _raw-function somehow.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
>

pboettch commented:
For the documentation part you can inspire yourself with what's done in `modbus_tcp_connect()`. 

For the testing part, ideally we'd also support rtutcp-server here in libmodbus (if that makes sense) then writing tests would be simpler.

v-zhuravlev commented:
But are there any plans to include this in the first place ? Implementing  server is overkill I think 

Отправлено с телефона

> 15 февр. 2018 г., в 18:12, Patrick Boettcher <notifications@github.com> написал(а):
> 
> For the documentation part you can inspire yourself with what's done in modbus_tcp_connect().
> 
> For the testing part, ideally we'd also support rtutcp-server here in libmodbus (if that makes sense) then writing tests would be simpler.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
>

pboettch commented:
I think it will be included. It's just a question of time.

i-ky committed:
Fixed MODBUS_GET_* macros in case of negative values

In case resulting value should be negative it is incorrect to use '+' operator to construct it from pieces, because highest bytes will result in negative number after bitwise shift while others will stay positive. Replacing addition with '|' should solve the issue.

myssta commented:
Please consider my approach (#445). It contains both server, client, test, documentation code. however I assumed that the slave id must be set and considered. Please comment especially whether the slave id should be considered or not. I assumed RTU considers it so RTU over TCP should do it, too.

strahlc commented:
any updates on this?

strahlc commented:
It is so annoying to keep adapting the patches to new versions of this library. 
Please merged this MR.
